routers/web/user/task.go

```diff
@@ -5,25 +5,51 @@ package user
 
 import (
 	"net/http"
-	"strconv"
 
 	admin_model "code.gitea.io/gitea/models/admin"
+	access_model "code.gitea.io/gitea/models/perm/access"
+	"code.gitea.io/gitea/models/unit"
 	"code.gitea.io/gitea/modules/json"
+	"code.gitea.io/gitea/modules/log"
 	"code.gitea.io/gitea/services/context"
 )
 
 // TaskStatus returns task's status
 func TaskStatus(ctx *context.Context) {
-	task, opts, err := admin_model.GetMigratingTaskByID(ctx, ctx.PathParamInt64("task"), ctx.Doer.ID)
+	task, _, err := admin_model.GetMigratingTaskByID(ctx, ctx.PathParamInt64("task"), 0)
```

Comment:
The logic is changes.

The old logic means "only creator could see the migration progress", but there is a bug: when doerID == 0, it passes all the check.

The new logic means: every one has read-access could see the migration progress.

----

To be honest, I think the old logic more reasonable, and it only needs one line to fix: `Eq{"doer_id": doerID}`.

Reply:
Currently, the logic is in the middle. The page can now be viewed if the repository is public, but the page has javascript error because the async request has a different permission check with the page.

We can have two chooses, one is migrating repository can only be viewed by the creator, another is migrating repository can be viewed according to the repository's permissions. This PR chose the later. But I'm not against the first one.

`ctx.PathParamInt64("task")` could be 0 or any digital, so we will check the permissions after this loading.

Reply:
4ac892d41bb5c691250bd5c065e8a68ea74cdabe

Reply:
That's my what I meant .... I mean: most new code could be removed, just do `Eq{task_id, doer_id}` in `GetMigratingTaskByID`.

Reply:
* https://github.com/go-gitea/gitea/pull/32152